pydal/objects.py

```diff
@@ -570,11 +576,18 @@ def __setitem__(self, key, value):
                 raise SyntaxError(
                     'value must be a dictionary: %s' % value)
             self.__dict__[str(key)] = value
+            if isinstance(value, (FieldVirtual, FieldMethod)):
```

Comment:
I've made this change because in tests we are defining virtual fields just setting them as attributes. I thought it was a feature. If not, we should change tests before.